core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -60,21 +64,26 @@
 @Service
 public class SpecService {
 
+  private final int SPEC_PUBLISHING_TIMEOUT_SECONDS = 5;
+
   private final FeatureSetRepository featureSetRepository;
   private final ProjectRepository projectRepository;
   private final StoreRepository storeRepository;
   private final Source defaultSource;
+  private final KafkaTemplate<String, FeatureSetProto.FeatureSetSpec> specPublisher;
 
   @Autowired
   public SpecService(
       FeatureSetRepository featureSetRepository,
       StoreRepository storeRepository,
       ProjectRepository projectRepository,
-      Source defaultSource) {
+      Source defaultSource,
+      KafkaTemplate<String, FeatureSetProto.FeatureSetSpec> specPublisher) {
```

Comment:
I just want to flag that even though I agree with the approach we are taking with Spring and Kafka, I do consider it technical debt that we are going into. Ideally the life cycle of jobs and the updates of feature sets to those jobs would fully encapsulated in the job management layer, especially if we ever want to separate job management from core.

Reply:
I would be more than happy to move this communication responsibility to JobService, but right now JobService is dependant on SpecService and I need publishing to kafka be synchronous part of applyFeatureSet. So yeah, currently it's a tech debt.